go/host/p2p/no_inbound_p2p.go

```diff
@@ -0,0 +1,263 @@
+package p2p
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"math/big"
+	"net"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/obscuronet/go-obscuro/go/common"
+	"github.com/obscuronet/go-obscuro/go/common/host"
+	"github.com/obscuronet/go-obscuro/go/common/log"
+	"github.com/obscuronet/go-obscuro/go/common/retry"
+	"github.com/obscuronet/go-obscuro/go/common/subscription"
+	"github.com/obscuronet/go-obscuro/go/config"
+
+	gethlog "github.com/ethereum/go-ethereum/log"
+)
+
+type NoInboundP2P struct {
```

Comment:
how much of this is duplicated compared to the normal implementation?
If a lot, then it would make sense to initialise that with a flag

Reply:
Yeah, I like using a flag, did that in the p2p in-memory; was keeping them separate to make sure I wasn't introducing any bugs in the existing p2p. 

I'll push all under a flag tomorrow 👍